### PR TITLE
CAMEL-17429: removed JDK 9 profiles

### PR DESCRIPTION
**Pull request by orpiske**
Obs.: there's more, but I am separating in smaller PRs to make it easier to review.

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: core/camel-core-engine/pom.xml [hunk 1]
```diff
@@ -349,43 +357,6 @@
...
-                            <excludes>
-                                <exclude>org/apache/camel/component/file/stress/**.java</exclude>
-                                <exclude>**/DistributedCompletionIntervalTest.java</exclude>
-                                <exclude>**/DistributedConcurrentPerCorrelationKeyTest.java</exclude>
-                                <exclude>${platform.skip.tests}</exclude>
-                            </excludes>
-                            <forkedProcessTimeoutInSeconds>3000</forkedProcessTimeoutInSeconds>
-                            <!-- needed for testing the properties component -->
-                            <environmentVariables>
-                                <FOO_SERVICE_HOST>myserver</FOO_SERVICE_HOST>
```
Comment (davsclaus) [inline core/camel-core-engine/pom.xml:381]: I think these ENV are used during test of camel-core. Not sure if we set them in another way.

File: core/camel-management-api/pom.xml [hunk 2]
```diff
@@ -242,42 +249,5 @@
...
-                            <excludes>
-                                <exclude>org/apache/camel/component/file/stress/**.java</exclude>
-                                <exclude>**/DistributedCompletionIntervalTest.java</exclude>
-                                <exclude>**/DistributedConcurrentPerCorrelationKeyTest.java</exclude>
-                                <exclude>${platform.skip.tests}</exclude>
-                            </excludes>
-                            <forkedProcessTimeoutInSeconds>3000</forkedProcessTimeoutInSeconds>
-                            <!-- needed for testing the properties component -->
-                            <environmentVariables>
-                                <FOO_SERVICE_HOST>myserver</FOO_SERVICE_HOST>
```
Comment (davsclaus) [inline core/camel-management-api/pom.xml:274]: Same as prev comment

File: parent/pom.xml [hunk 3]
```diff
@@ -4040,6 +4040,68 @@
         </dependencies>
     </dependencyManagement>
 
+    <dependencies>
```
Comment (davsclaus) [inline parent/pom.xml:4043]: We should in another PR see what of this crap we can get rid of

File: parent/pom.xml [hunk 4]
```diff
@@ -4048,6 +4110,13 @@
                     <groupId>org.apache.camel</groupId>
                     <artifactId>camel-package-maven-plugin</artifactId>
                     <version>${project.version}</version>
+                    <dependencies>
```
Comment (davsclaus) [inline parent/pom.xml:4113]: Oh maybe this jaxb dependency should be added directly to camel-package-maven-plugin so its not needed to be added in this special way

File: core/camel-core-engine/pom.xml (hunk 1 above)
Comment (orpiske) [reply core/camel-core-engine/pom.xml:381]: We should be fine, `core/camel-core-engine` does not have any test in itself. The tests are in the `core/camel-core`  module which already define them.

File: core/camel-management-api/pom.xml (hunk 2 above)
Comment (orpiske) [reply core/camel-management-api/pom.xml:274]: We should be fine, `core/camel-core-engine` does not have any test in itself. The tests are in the `core/camel-core`  module which already define them.

File: parent/pom.xml (hunk 3 above)
Comment (orpiske) [reply parent/pom.xml:4043]: +1. I created CAMEL-17435 so I don't forget to review those (for now, I mostly did a 1-to-1 translation of the jdk9+ profiles that were activated by default). I'll take care of that tomorrow, after I come back from a short PTO.

File: parent/pom.xml (hunk 4 above)
Comment (orpiske) [reply parent/pom.xml:4113]: Oh, good idea. I think we can remove this, since it's already defined as a dependency in that module.

**Review by davsclaus — APPROVED** [review]

**Pull request merged**